skopt/space.py

```diff
@@ -246,13 +270,43 @@ def __init__(self, categories, prior=None):
         self.categories = categories
         self.transformer = _CategoricalEncoder()
         self.transformer.fit(self.categories)
+        self.prior = prior
 
         if prior is None:
             prior = np.tile(1. / len(self.categories), len(self.categories))
```

Comment:
It should be easier to introduce a `self.prior_` attribute and equate that to the `np.tile(` here. In that way you can directly use `np.allclose` to compare `self.prior_` with `other.prior_` and do away with the `priors_are_equal` method.